src/DependencyInjection/Configuration.php

```diff
@@ -15,122 +15,122 @@ class Configuration implements ConfigurationInterface
      */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('printdeal_pandosearch');
+        $treeBuilder = new TreeBuilder('printdeal_pandosearch', 'array');
 
+        /** @var ArrayNodeDefinition $rootNode */
+        $rootNode = $treeBuilder->getRootNode();
```

Comment:
we can replace this line with setRoot node to save BC
```suggestion
        $rootNode = $treeBuilder->root('printdeal_pandosearch');
```